src/EditorFeatures/CSharp/MisplacedUsingDirectives/MisplacedUsingDirectivesCodeFixProvider.cs

```diff
@@ -0,0 +1,394 @@
+
+            // To expand the using declarations in the namespace we need to get an updated document, since the 
+            // expander requires that the SyntaxTree for the semantic model match that of the using directives.
+            var newDocument = document.WithSyntaxRoot(compilationUnitWithExpandedUsings);
+            var newCompilationUnit = (CompilationUnitSyntax)await newDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // Expand the using declarations in the namespace declaration.
+            var namespaceDeclaration = newCompilationUnit.Members.OfType<NamespaceDeclarationSyntax>().First();
+
+            var expandNamespaceDeclarationUsingsTasks = namespaceDeclaration.Usings.Select(
+                usingDirective => ExpandUsingDirectiveAsync(newDocument, usingDirective, cancellationToken));
+            var newNamespaceDeclarationUsings = await Task.WhenAll(expandNamespaceDeclarationUsingsTasks).ConfigureAwait(false);
+
+            var namespaceDeclarationWithExpandedUsings = namespaceDeclaration.WithUsings(new SyntaxList<UsingDirectiveSyntax>(newNamespaceDeclarationUsings));
+
+            return newCompilationUnit.ReplaceNode(namespaceDeclaration, namespaceDeclarationWithExpandedUsings);
+        }
+
+        private static async Task<UsingDirectiveSyntax> ExpandUsingDirectiveAsync(Document document, UsingDirectiveSyntax usingDirective, CancellationToken cancellationToken)
+        {
+            var newName = await Simplifier.ExpandAsync(usingDirective.Name, document, cancellationToken: cancellationToken).ConfigureAwait(false);
+            return usingDirective.WithName(newName);
+        }
+
+        private static CompilationUnitSyntax MoveUsingsInsideNamespace(CompilationUnitSyntax compilationUnit)
+        {
+            // Get the compilation unit usings and set them up to format when moved.
+            var usingsToAdd = compilationUnit.Usings.Select(
+                directive => directive.WithAdditionalAnnotations(Formatter.Annotation));
+
+            // Remove usings and fix leading trivia for compilation unit.
+            var compilationUnitWithoutUsings = compilationUnit.WithUsings(new SyntaxList<UsingDirectiveSyntax>());
+            var compilationUnitWithoutBlankLine = RemoveLeadingBlankLinesFromFirstMember(compilationUnitWithoutUsings);
+
+            // Fix the leading trivia for the namespace declaration.
+            var namespaceDeclaration = compilationUnitWithoutBlankLine.Members.OfType<NamespaceDeclarationSyntax>().First();
+            var namespaceDeclarationWithBlankLine = EnsureLeadingBlankLineBeforeFirstMember(namespaceDeclaration);
+
+            // Update the namespace declaration with the usings from the compilation unit.
+            var newUsings = namespaceDeclarationWithBlankLine.Usings.InsertRange(0, usingsToAdd);
+            var namespaceDeclarationWithUsings = namespaceDeclarationWithBlankLine.WithUsings(newUsings);
```

Comment:
if the usings were originally sorted, can we keep them sorted? i believe we have helpers for that.

Reply:
I'll take a look at it tomorrow. I know the previous version of this Fixer did too much (group, sort, move) and we were shooting for something that did just the move part, with the assumption that this could be shared with Organize Imports. I did have a version of the fixer that used the AddImportService which would insert them into the sorted position, however, trivia could be funky at times.

Reply:
I think sorting might be important when user does a FixAll across project/solution. The only question would be whether it is common for users to have a mix of usings inside and outside to hit this case often enough to warrant this functionality.